syncer/sync.py

```diff
@@ -22,52 +17,63 @@
 print("Done!")
 
 print("Connecting to Pachyderm")
-pc = pach.Client(host=os.environ["PACHD_SERVICE_HOST"], port=os.environ["PACHD_SERVICE_PORT"])
+PACHY_CLIENT = pach.Client(
+    host=os.environ["PACHD_SERVICE_HOST"], port=os.environ["PACHD_SERVICE_PORT"]
+)
 print("Done!")
 
 # Specify "files" to create File datasets (matches all files in the Pachyderm
 # repo), or "jsonl" for Tabular json lines format (matches **/*.jsonl in the
 # Pachyderm repo).
 MODES = {"files", "jsonl", "delimited"}
-mode = os.getenv("PACHYDERM_SYNCER_MODE", "jsonl")
-if mode not in MODES:
+MODE = os.getenv("PACHYDERM_SYNCER_MODE", "jsonl")
+if MODE not in MODES:
     raise ValueError(f"mode must be one of {MODES}")
 
 # 1. Get all the repos
 # 2. For each repo, get all the commits
 # 3. For each (repo, commit) see if there's a (dataset, version) that points to it
 # 4. If not, create the new (dataset, version)
-
-# TODO: don't create a new version for every existing version every startup
-# (parse list of existing, filter them out)
+# Track Pachyderm commits that have been propagated
+EXISTING_COMMITS = set()
```

Comment:
Is it a convention in python to capitalize non-constant globals? In e.g. Go I think the convention is that all caps=constant but I don't know if that's universal.

Reply:
For reference. https://www.python.org/dev/peps/pep-0008/#constants

Here `EXISTING_COMMITS` is a `const` since we never override it with a different set.

I do admit that sometimes I unconsciously bring conventions from Go to Python tho 😅

Reply:
The code does mutate it though